orchestra/analytics/latency.py

```diff
@@ -0,0 +1,112 @@
+    # access all over the database.  Ideally, we'd have a sequential scan
+    # version of this.
+    time_data = list(itertools.chain.from_iterable(
+        project_time_row_generator(
+            get_project_information(project.id))
+        for project in projects))
+    df = DataFrame(time_data)
+    df = df[
+        ['project', 'task', 'assignment', 'iteration', 'start_datetime',
+         'calendar_time', 'work_time']]
+    return df
+
+
+class Iteration(object):
+    """ Represents a task assignment iteration's data for reporting. """
+    def __init__(self, project, task, assignment, iteration, work_time,
+                 end_datetime):
+        self.project = project
+        self.task = task
+        self.assignment = assignment
+        self.iteration = iteration
+        self.work_time = work_time
+        self.end_datetime = end_datetime
+
+    def to_dict(self, start_datetime):
+        return {
+            'project': self.project['id'],
+            'task': self.task['id'],
+            'assignment': self.assignment,
+            'iteration': self.iteration,
+            'start_datetime': start_datetime,
+            'work_time': self.work_time,
+            'calendar_time': self.end_datetime - start_datetime
+        }
+
+
+def project_time_row_generator(project_information):
+    """
+    Given a project, yields all task assignment iterations, reporting
+    calendar and work times.
+    """
+    # lol at this variable name
+    human_slugs = {
+        step['slug'] for step in project_information['steps']
+        if step['worker_type'] == Step.WorkerType.HUMAN}
+
+    for step_slug, task in iter(project_information['tasks'].items()):
+        if step_slug not in human_slugs:
+            continue
+        if task['status'] != dict(Task.STATUS_CHOICES)[Task.Status.COMPLETE]:
+            continue
+        iterations = []
+        for assignment_idx, assignment in enumerate(task['assignments']):
+            for iteration, snapshot in enumerate(
+                    assignment['snapshots']['snapshots']):
+                if iteration == 0:
+                    iterations.append(Iteration(
+                        project_information['project'], task,
+                        assignment_idx, 0, timedelta(seconds=0),
+                        parse(assignment['start_datetime'])))
+                iterations.append(Iteration(
+                    project_information['project'], task,
+                    assignment_idx,
+                    iteration + 1,
+                    timedelta(seconds=snapshot['work_time_seconds']),
+                    parse(snapshot['datetime'])))
+        iterations = sorted(iterations, key=attrgetter('end_datetime'))
+        iteration_times = (
+            [parse(task['start_datetime'])] +
+            [iteration.end_datetime for iteration in iterations])
+        for start_datetime, iteration in zip(iteration_times[:-1], iterations):
```

Comment:
couldn't this just be:

    start_datetime = parse(task['start_datetime'])
    for iteration in iterations:
        yield iteration.to_dict(start_datetime)